### PR TITLE
First and last arguments must be positive integers

### PR DESCRIPTION
**Pull request by vslinko**
Ref https://github.com/facebook/relay/issues/735

### CODE REVIEW

**Comment by wincent** [comment]
Thanks for doing this, @vslinko, and sorry about the delay in commenting on it. I think that as per our discussion on https://github.com/facebook/relay/issues/735 we should go with "non-negative" rather than "positive".

**Comment by vslinko** [comment]
@wincent updated

**Comment by wincent** [comment]
Thanks for the update, @vslinko. I see you've changed the wording here but not the semantics. Shouldn't the conditional change to `< 0` instead of `<= 0` as well?

Also, will comment with one tiny nit inline.

File: src/connection/arrayconnection.js [hunk 1]
```diff
@@ -85,12 +85,20 @@ export function connectionFromArraySlice<T>(
...
     endOffset = Math.min(
       endOffset,
       startOffset + first
     );
   }
   if (last != null) {
+    if (last <= 0) {
+      throw new Error('Argument "last" must be an non-negative integer');
+    }
+
```
Comment (wincent) [inline src/connection/arrayconnection.js]: For both of these I think we want `a non-negative`, not `an non-negative`.

**Comment by vslinko** [comment]
@wincent fixed

**Comment by wincent** [comment]
Think this needs a rebase, and Travis is currently unhappy with it.

**Comment by vslinko** [comment]
@wincent rebased

**Comment by wincent** [comment]
Thanks!

**Pull request merged**